pkg/plugins/findDefaults.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package plugins
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/kardianos/osext"
+	"github.com/shirou/gopsutil/process"
+)
+
+var (
+	// a list of directories to scan for potential location
+	// of avalanchego configs
+	scanConfigDirs = []string{}
+	// env var for avalanchego data dir
+	defaultUnexpandedDataDir = "$" + config.AvalancheGoDataDirVar
+	// expected file name for the config
+	// TODO should other file names be supported? e.g. conf.json, etc.
+	defaultConfigFileName = "config.json"
+	// expected name of the plugins dir
+	defaultPluginDir = "plugins"
+	// default dir where the binary is usually found
+	defaultAvalanchegoBuildDir = filepath.Join("go", "src", "github.com", constants.AvaLabsOrg, constants.AvalancheGoRepoName, "build")
+)
+
+// this init is partly "borrowed" from avalanchego/config/config.go
+func init() {
```

Comment:
Why putting this in init and not in a function that generates and return the scanConfigDirs?

Reply:
I copied this from old code, but I reworked this into a new method and called that from our initial setup in the root command.

Reply:
I mean there is no need for a scanConfigDirs global variable, the other functions can just call a version of SetScanConfigDirs that return the dirs. So there is no needed to fill up the variable on init (or is there other convenience on doing it?)